### PR TITLE
CHAOSPLT-64: Add a Spec.SetDefaults method

### PR DESCRIPTION
**Pull request by ptnapoleon**
We have a desire to set the defaults directly on a DisruptionSpec. So I've extracted a SetDefaults function that is called by the webhook's Default function, leading to no change the chaos-controller's functionality.

Unfortunately, this makes the logging a bit trickier? Do we want to stop logging when we set defaults? I don't want to pass a logger, along with the disruption name and namespace into the SetDefaults function. Any ideas?

### CODE REVIEW

File: api/v1beta1/disruption_webhook.go [hunk 1]
```diff
@@ -114,8 +114,9 @@ var _ webhook.Defaulter = &Disruption{}
 func (r *Disruption) Default() {
 	if r.Spec.Duration.Duration() == 0 {
 		logger.Infow(fmt.Sprintf("setting default duration of %s in disruption", defaultDuration), "instance", r.Name, "namespace", r.Namespace)
```
[the remark this answers is not in the document]
Comment (clairecng) [reply api/v1beta1/disruption_webhook.go:116]: oooh one less log line with a namespace tag, count me in

[the remark this answers is not in the document]
Comment (clairecng) [reply api/v1beta1/disruption_webhook.go:116]: We already know the default duration on the logling of the chaos-controller startup right?

[the remark this answers is not in the document]
Comment (clairecng) [reply api/v1beta1/disruption_webhook.go]: hm ok
I think it's ok to remove it. If there's a bug there I'm pretty sure we'll notice it right away, with tests and all?

**Review by diyarab — APPROVED** [review]
lgtm

**Pull request merged**